mono/profiler/log.c

```diff
@@ -1200,32 +1200,64 @@ gc_reference (MonoObject *obj, MonoClass *klass, uintptr_t size, uintptr_t num,
 }
 
 static void
-gc_roots (MonoProfiler *prof, MonoObject *const *objects, const MonoProfilerGCRootType *root_types, const uintptr_t *extra_info, uint64_t num)
+gc_roots (MonoProfiler *prof, uint64_t num, const mono_byte *const *addresses, const MonoObject* const *objects)
 {
 	ENTER_LOG (&heap_roots_ctr, logbuffer,
 		EVENT_SIZE /* event */ +
 		LEB128_SIZE /* num */ +
-		LEB128_SIZE /* collections */ +
 		num * (
-			LEB128_SIZE /* object */ +
-			LEB128_SIZE /* root type */ +
-			LEB128_SIZE /* extra info */
+			LEB128_SIZE /* address */ +
+			LEB128_SIZE /* object */
 		)
 	);
 
 	emit_event (logbuffer, TYPE_HEAP_ROOT | TYPE_HEAP);
 	emit_value (logbuffer, num);
-	emit_value (logbuffer, mono_gc_collection_count (mono_gc_max_generation ()));
 
 	for (int i = 0; i < num; ++i) {
-		emit_obj (logbuffer, objects [i]);
-		emit_byte (logbuffer, root_types [i]);
-		emit_value (logbuffer, extra_info [i]);
+		emit_ptr (logbuffer, addresses [i]);
+		emit_obj (logbuffer, (void*)objects [i]);
 	}
 
 	EXIT_LOG;
 }
 
+static void
+gc_root_register (MonoProfiler *prof, const mono_byte *start, size_t size, MonoGCRootSource kind, const void *key, const char *msg)
+{
+	int msg_len = msg ? strlen (msg) + 1 : 0;
+	ENTER_LOG (&heap_roots_ctr, logbuffer,
```

Comment:
I forgot to mention this in my last review: Each kind of event should have its own kind of counter, so you'll want to add one for root register and root unregister events.